src/main/resources/reference.conf

```diff
@@ -123,6 +175,230 @@ resourceTypes = {
     authDomainConstrainable = true
     reuseIds = false
   }
+  controlled-user-shared-workspace-resource = {
+    actionPatterns = {
+      delete = {
+        description = "delete this resource"
+      }
+      read_policies = {
+        description = "view all policies and policy details for this resource"
+      }
+      set_parent = {
+        description = "set the parent workspace of this resource"
+      }
+      "share_policy::owner" = {
+        description = "change the membership of the owner policy for this resource"
+      }
+      "share_policy::editor" = {
+        description = "change the membership of the editor policy for this resource"
+      }
+      "share_policy::writer" = {
+        description = "change the membership of the writer policy for this resource"
+      }
+      "share_policy::reader" = {
+        description = "change the membership of the reader policy for this resource"
+      }
+      "read" = {
+        description = "perform reader actions on the resource"
+      }
+      "write" = {
+        description = "perform writer actions on the resource"
+      }
+      "edit" = {
+        description = "perform editor actions on the resource"
+      }
+      "own" = {
+        description = "perform owner actions on the resource"
+      }
+    }
+    ownerRoleName = "owner"
+    roles = {
+      owner = {
+        roleActions = ["delete", "read_policies", "share_policy::owner", "share_policy::editor", "share_policy::writer", "share_policy::reader", "own", "set_parent"]
```

Comment:
Does owner need edit? I think it is missed in all cases below.

Reply:
iiuc the owner role on a resource is only granted to the WSM service account - it definitely needs delete permission, but I'm less sure about edit. Added for now, but that might change